src/clients/add/SamlConfig.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import {
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Switch,
+} from "@patternfly/react-core";
+
+import type { ClientForm } from "../ClientDetails";
+import { FormAccess } from "../../components/form-access/FormAccess";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+export const Toggle = ({ name, label }: { name: string; label: string }) => {
+  const { t } = useTranslation("clients");
+  const { control } = useFormContext<ClientForm>();
+
+  return (
+    <FormGroup
+      hasNoPaddingTop
+      label={t(label)}
+      fieldId={label}
+      labelIcon={
+        <HelpItem
+          helpText={t(`clients-help:${label}`)}
+          forLabel={t(label)}
+          forID={label}
+        />
+      }
+    >
+      <Controller
+        name={name}
+        defaultValue="false"
+        control={control}
+        render={({ onChange, value }) => (
+          <Switch
+            id={name!}
+            data-testid={label}
+            label={t("common:on")}
+            labelOff={t("common:off")}
+            isChecked={value === "true"}
+            onChange={(value) => onChange("" + value)}
+          />
+        )}
+      />
+    </FormGroup>
+  );
+};
+
+export const SamlConfig = () => {
+  const { t } = useTranslation("clients");
+  const { control } = useFormContext<ClientForm>();
+
+  const [nameFormatOpen, setNameFormatOpen] = useState(false);
+  return (
+    <FormAccess
+      isHorizontal
+      role="manage-clients"
+      className="keycloak__capability-config__form"
+    >
+      <FormGroup
+        label={t("nameIdFormat")}
+        fieldId="nameIdFormat"
+        labelIcon={
+          <HelpItem
+            helpText="clients-help:nameIdFormat"
+            forLabel={t("nameIdFormat")}
+            forID={t("common:helpLabel", { label: t("nameIdFormat") })}
+          />
+        }
+      >
+        <Controller
+          name="attributes.saml_name_id_format"
+          defaultValue="username"
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="samlNameIdFormat"
+              onToggle={(open) => setNameFormatOpen(open)}
+              onSelect={(_, value) => {
+                onChange(value as string);
```

Comment:
Since `SelectOptionObject` implements a `toString()` method there is no need to type-cast this value.
```suggestion
                onChange(value.toString());
```